drones/image_processing/yolo.py

```diff
@@ -0,0 +1,138 @@
+
+        # Convert
+        img = img[:, :, ::-1].transpose(2, 0, 1)  # BGR to RGB, to 3x416x416
+        img = np.ascontiguousarray(img)
+
+        # Get names and colors
+        names = self.model.module.names if hasattr(self.model, "module") else self.model.names
+
+        device = select_device(self.device_type)
+
+        # Run inference
+        if device.type != "cpu":
+            self.model(
+                torch.zeros(1, 3, img_size, img_size).to(device).type_as(next(self.model.parameters()))
+            )  # run once
+        t0 = time.time()
+        result = []
+        image = torch.from_numpy(img).to(device)
+        image = image.float()  # uint8 to fp16/32
+        image /= 255.0  # 0 - 255 to 0.0 - 1.0
+        if image.ndimension() == 3:
+            image = image.unsqueeze(0)
+
+        # Inference
+        pred = self.model(image)[0]
+
+        # Apply NMS
+        pred = non_max_suppression(pred, self.conf_tres, self.iou_thres, classes=self.classes)
+
+        # Process detections
+        for i, det in enumerate(pred):  # detections per image
+            gn = torch.tensor(img0.shape)[[1, 0, 1, 0]]  # normalization gain whwh
+            if len(det):
+                # Rescale boxes from img_size to im0 size
+                det[:, :4] = scale_coords(image.shape[2:], det[:, :4], img0.shape).round()
+
+                # Write results
+                for *xyxy, conf, cls in reversed(det):
+                    xywh = (xyxy2xywh(torch.tensor(xyxy).view(1, 4)) / gn).view(-1).tolist()  # normalized xywh
+                    line = (names[int(cls)], *xywh)  # label format
+                    result.append(line)
+
+        logging.getLogger(__name__).info(f"Processed image, processing time: ({time.time() - t0:.3f}s)")
+        return result
+
+    def detect_object_yolo(self, image: np.ndarray) -> list:
+        """Function will detect objects on given image and return position and width of object, which class is chosen
+        in config file. Many parameters of detection such as weights, thresholds and others can be set inside config
+        file. In case of many objects detected, all of them will be returned.
+
+        Parameters:
+        ----------
+            img0: np.ndarray
+                image on which objects detection will be proceeded
+
+        Returns:
+        ----------
+            center_and_diameter: list of tuples(int, int, int)
+                Every found instance of object defined in config file has it's own tuple. Every tuple have 3 values,
+                which represent object center coordinates(x,y) and width.
+                If the object is not detected list will be empty.
+        """
+
+        # Using yolo detect object on photo
+        results = self.detect(image)
+
+        image_width = image.shape[1]
+        image_height = image.shape[0]
+
+        result_list = []
+        # Find proper class on photo. Should be only one represent of this class
```

Comment:
I don't get this comment, but I guess it's not right, as we are collecting all instances of the class. Could you explain what you meant?